x/subscription/README.md

```diff
@@ -0,0 +1,202 @@
+
+There could only be a single active subscription for a consumer.  
+When a consumer purchase a subscription, the tokens are transferred from their account immediately.  
+Those tokens are later on distributed among the providers that served that consumer, and the delegators that staked those providers.
+
+### Advance Month
+
+One of the primary processes within this module involves the function `advanceMonth`, which is invoked upon the expiration of a timer set by the subscription. This function holds significance as it manages the logic that occurs at the conclusion of each month of a subscription.
+
+Here's a high-level flow of the function:
+
+1. **Update the CU Tracker Timer:**
+
+   - Add a CU tracker timer for the subscription.
+
+2. **Check Subscription Duration:**
+
+   - If the subscription’s remaining duration (`DurationLeft`) is zero:
+     - If this occurs, it's considered a bug. Log the issue and extend the subscription by one month for smoother operation and investigation.
+   - If there is still time left (`DurationLeft` > 0):
+     - reduce it by one to reflect the passage of a month.
+     - Increase the total duration (`DurationTotal`) by one.
+     - Reset and update the subscription details.
+   - If no time is left (`DurationLeft` = 0):
+     - Subscription is expired. Refer to the next step.
+
+3. **Handle Special Cases for Subscriptions:**
+
+   - If a future subscription is set:
+     - Activate the future subscription by updating current subscription details with those of the future subscription and restart the subscription.
+   - Else, if auto-renewal is enabled:
+     - Attempt to renew the subscription for another month.
+     - If renewal fails, remove the expired subscription.
+   - If there is no future subscription or auto-renewal, remove the expired subscription.
+
+### Subscription Upgrade
+
+A subscription can be upgraded to a more expensive plan.
+
+That also means, that if a consumer buys a plan, and then decides to downgrade their plan, the way to do it is either buy more months in the cheaper subscription, so the amount that the consumer will pay will be larger than the one using the same command as shown above, with the only caveat - the `plan-index` must be different than the currently active subscription's plan, and, it must be with a higher price.
+
+Tokens are deducted from the creator's account immediately, and the new plan becomes effective in the next epoch.  
+The old subscription will still be saved in the fixation store, for buffering purposes (the amount of blocks that the subscription will be saved for, is determined in the `BlocksToSave` function, under the [EpochStorage](https://github.com/lavanet/lava/blob/main/x/epochstorage/README.md) module).
+
+More ways to upgrade are by making an [Advance Purchase](#advance-purchase) or enabling [Auto Renewal](#auto-renewal).
+
+### Subscription Renewal
```

Comment:
why separate subscription renewal and auto renewal? put them under the same section

Reply:
Combined in 7ef2ecf36129e8175e2ff9e0dd04eef924e6d1f2